Sources/PackageModel/PackageIdentity.swift

```diff
@@ -149,6 +160,11 @@ extension PackageIdentity {
             self = scope
         }
 
+        fileprivate init?(_ substring: String.SubSequence?) {
+            guard let substring = substring else { return nil }
+            self.init(String(substring))
+        }
```

Comment:
`let identity = sequence.map{ String($0) }` at the call site too verbose?

Reply:
The alternative to this is:

```swift
guard components.count == 2,
              let scope = components.first.flatMap({ Scope(String($0)) }),
              let name = components.last.flatMap({ Name(String($0)) })
        else { return nil }
```

This seemed to me less clear, but really it's a subjective call. Totally up to you which convention you'd prefer to have in the code base.